server.js

```diff
@@ -168,21 +163,22 @@ LimitdServer.prototype.stop = function (callback) {
     } else {
       logger.debug({ address }, 'server closed');
     }
-    this._db.close(dbCloseError => {
+    this._db.close((dbCloseError) => {
       if (dbCloseError) {
         logger.error({
           err: dbCloseError
         }, 'error closing the database');
       } else {
         logger.debug('database closed');
       }
-      self.emit('close');
+      this.emit('close');
       return callback(serverCloseError || dbCloseError);
     });
   });
-
 };
 
+LimitdServer.prototype.updateBucketConfig = function (buckets) {
+  this._db.loadTypes(buckets);
```

Comment:
Have you tested this under load to see how the server behaves? Do we fail all requests while this reload, or we wait...?

Reply:
Expected behavior is we wait.